plugin/pkg/auth/authenticator/gssapi/gssproxy/gssproxy.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gssproxy
+
+import (
+	"errors"
+	"net"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/auth/authenticator"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/auth/user"
+	"github.com/nalind/gss/pkg/gss/proxy"
+)
+
+type GssProxyAuthenticator struct {
+	proxyPath string
+}
+
+func NewGssProxy(proxyPath string) (authenticator.Negotiate, error) {
+	return &GssProxyAuthenticator{
+		proxyPath: proxyPath,
+	}, nil
+}
+
+func (a *GssProxyAuthenticator) AuthenticateNegotiate(token []byte) (user.Info, bool, error) {
+	var callCtx proxy.CallCtx
+	var secCtx proxy.SecCtx
+	if len(token) == 0 {
+		return nil, false, errors.New("Requested Negotiate auth, but provided no auth data")
+	}
+	conn, err := net.Dial("unix", a.proxyPath)
+	if err != nil {
+		return nil, false, err
+	}
+	defer conn.Close()
+	gccr, err := proxy.GetCallContext(&conn, &callCtx, nil)
+	if err != nil {
+		return nil, false, err
+	}
+	if gccr.Status.MajorStatus != proxy.S_COMPLETE {
+		return nil, false, errors.New("Error returned from gss-proxy GetCallContext function: " + gccr.Status.MajorStatusString + "/" + gccr.Status.MinorStatusString)
+	}
+	ascr, err := proxy.AcceptSecContext(&conn, &callCtx, &secCtx, nil, token, nil, false, nil)
+	if err != nil {
+		return nil, false, err
+	}
+	if secCtx.NeedsRelease {
+		defer proxy.ReleaseSecCtx(&conn, &callCtx, &secCtx)
+	}
+	if ascr.Status.MajorStatus == proxy.S_CONTINUE_NEEDED {
+		// We can't handle multiple-round-trip requests right.
+		return nil, false, errors.New("Negotiate authentication canceled (incomplete): " + ascr.Status.MajorStatusString + "/" + ascr.Status.MinorStatusString)
```

Comment:
Right, but kubectl isn't the type of client to do that, since its authentication configuration's pre-specified.  That's more of a browser (and when I tried it, curl) case.

Being able to accept auth via a proxy doesn't seem to require that the server be unable to handle some things by itself.  We'd need an additional plugin to accept user identity information provided by a proxy, though.